crypto/conf/conf_ssl.c

```diff
@@ -35,12 +36,25 @@ struct ssl_conf_cmd_st {
     char *arg;
 };
 
+static CRYPTO_ONCE init_ssl_names_lock = CRYPTO_ONCE_STATIC_INIT;
+static CRYPTO_RWLOCK *ssl_names_lock;
 static struct ssl_conf_name_st *ssl_names;
 static size_t ssl_names_count;
 
-static void ssl_module_free(CONF_IMODULE *md)
+DEFINE_RUN_ONCE_STATIC(do_init_ssl_names_lock)
+{
+    ssl_names_lock = CRYPTO_THREAD_lock_new();
```

Comment:
This will leak the lock on exit won't it?

Reply:
Shouldn't do. It is freed by `ssl_module_free` which should be automatically called on exit.

Reply:
Oh...actually, no - its not freed by `ssl_module_free`. I was misreading the code

Reply:
ssl_names_lock will remain but I have no idea how to free it. I would make sense to make it static without the pointer…
ssl_module_free() frees the string but I can't free the lock pointer there since other threads might block on it.

Reply:
I think we should wait with this PR until the question in #19248 is resolved. There are a couple of different solutions to the problem above that spring to mind - but if we need to fix #19248 that may completely change things.